### PR TITLE
Editing and adding content in the code quality chapter

### PR DESCRIPTION
**Pull request by CapTen101**
### Summary

Addresses #1075 
@malvikasharan 

*This pull request is a small self-contained contribution to the Turing Way project under gsod 2020.*

### List of changes proposed in this PR (pull-request)

- Overall improved `Code Quality` section
- Static Code analysis section has a new table mentioning all widely used code analysis tools for major languages. Also added the advantages sub-section in the same.
- mentioned standard code guideline of Javascript.
- Introduced two different styles of auto code formatting. Wrote the JSON line need to turn on auto format in VScode(being one of the most famous text-editors out there).
- Fixed typos.
- Auto Formatting Code section has a new table mentioning all widely used beautify tools for different languages.

### What should a reviewer concentrate their feedback on?

- [x] Everything looks ok?

### Acknowledging contributors

- [x] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file:

### CODE REVIEW

File: book/website/code_quality/code_quality.md [hunk 1]
```diff
@@ -20,9 +23,28 @@ Tools for static code analysis can report bugs as well as style issues without e
 
 ## Static code analysis
 
-Static code analysis tools, also known as linters, can analyze source code and find bugs before your code is executed.
-In addition to finding bugs, many of these tools can also help maintain a consistent coding style.
-Examples of linter tools include [prospector](https://prospector.readthedocs.io) for Python, [lintr](https://github.com/jimhester/lintr) for R, or [shellcheck](https://www.shellcheck.net) for shell scripts.
+
+Static code analysis is a method that examines code and detects software vulnerabilities before before your code is executed or project is built and deployed. This analysis is capable of identifying quality issues, including security weaknesses and errors. In addition to finding bugs, many of these tools can also help maintain a consistent coding style.
```
Comment (malvikasharan) [inline book/website/code_quality/code_quality.md:27]:
```suggestion
Static code analysis is a method that examines code and detects software vulnerabilities before your code is executed or the project is built and deployed. 
This analysis is capable of identifying quality issues, including security weaknesses and errors. 
In addition to finding bugs, many of these tools can also help maintain a consistent coding style.
```

Comment (malvikasharan) [reply book/website/code_quality/code_quality.md:27]: Line breaks for each new line for consistency (defined in the style guide of the contributing file)

**Comment by CapTen101** [comment]
For sure! @malvikasharan I'll be doing it in half n hour or so. :smile:

**Comment by malvikasharan** [comment]
> For sure! @malvikasharan I'll be doing it in half n hour or so. 😄

No hurry at all!

**Comment by CapTen101** [comment]
Hi @malvikasharan! so i integrated the request changes to adhere to the style guide. Can you elaborate a little on the checklist part? like any example of it, to better understand what exactly I have to do. thanks! :smile:

**Comment by malvikasharan** [comment]
Our chapters have a structure as listed [here](https://github.com/alan-turing-institute/the-turing-way/blob/master/templates/CHAPTER_TEMPLATE.md), you can see examples in most chapters for the checklist, for example in the [Version Control chapter](https://the-turing-way.netlify.app/version_control/vcs-checklist.html).

**Comment by CapTen101** [comment]
Oh, right. Thanks, that resolves my query. I'm on it :+1:

**Comment by CapTen101** [comment]
Hi @malvikasharan I've made the necessary changes. I've added the checklist for Code Quality as well as the resources(bibliography, further reading) section. Kindly have a look :smile:

**Comment by malvikasharan** [comment]
HI @CapTen101. I think this looks good. I will tag one more reviewer to have a look and approve it. Thanks a lot for the fast action. 🎉

**Comment by CapTen101** [comment]
Sure! @malvikasharan. I'm happy to contribute to Turing Way project :smile:. I'm looking forward to contributing even more under GSoD 2020 :smile:

**Comment by CapTen101** [comment]
Hi @annakrystalli @crangelsmith Kindly review this PR :slightly_smiling_face:

**Comment by crangelsmith** [comment]
Hi! I think it looks good, also I learned something today by reading the section. Thanks!

**Comment by CapTen101** [comment]
I'm glad that it helped! :smile:

**Comment by malvikasharan** [comment]
Thanks a lot @crangelsmith for the quick response. 🎉

@all-contributors please add @CapTen101 for bug and content.

**Comment by CapTen101** [comment]
wow, thank you so much! @malvikasharan :smile:

**Comment by CapTen101** [comment]
@malvikasharan Should I now start making my draft proposal if you feel satisfied with my contribution as a task for GSoD ? :slightly_smiling_face:

**Comment by malvikasharan** [comment]
Hi Tushar, please note that there are multiple candidates at the moment contributing for demo and hence we will be waiting until the end of this week to make any selection with Kirstie Whitaker (project lead). Meanwhile, please drop me an email with a rough idea that you have in mind for GSoD.

**Comment by CapTen101** [comment]
ok, cool 👍

**Comment by CapTen101** [comment]
Hi @malvikasharan !
As you asked, I've sent you a mail with an attached doc having all the rough ideas i have in mind for the project and which will eventually form as the basis for my proposal. Kindly have a look at it :smile:

**Pull request merged**